lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx

```diff
@@ -914,6 +914,10 @@ void GeneralLinearModelAlgorithm::setNoise(const Point & noise)
 {
   const UnsignedInteger size = inputSample_.getSize();
   if (noise.getSize() != size) throw InvalidArgumentException(HERE) << "Noise size=" << noise.getSize()  << " does not match sample size=" << size;
+  // Currently setNoise is not handled with HMAT
+  // We should first rework on the hmat side to promote this possibility
+  if (getMethod() == 1)
+    throw InvalidArgumentException(HERE) << "Noise is not be handled with HMAT method" ;
```

Comment:
maybe notyetimplementedexception is more appropriate ?

Reply:
+1